searchcommon/src/vespa/searchcommon/attribute/i_attribute_functor.h

```diff
@@ -14,8 +16,14 @@ namespace search::attribute {
 class IAttributeFunctor
 {
 public:
-    virtual void operator()(const search::AttributeVector &attributeVector) = 0;
+    virtual void operator()(const IAttributeVector &attributeVector) = 0;
     virtual ~IAttributeFunctor() { }
 };
 
+class IAttributeExecutor {
+public:
+    virtual ~IAttributeExecutor() { }
+    virtual void asyncForAttribute(const vespalib::string &name, std::shared_ptr<IAttributeFunctor> func) const = 0;
```

Comment:
as far as I understand, the functor given here heavily relies on the actual type of the underlying attribute. maybe we should consider passing something more abstract here, like "++", and resolve for type when we are ready to apply it to the actual attribute vector.

Reply:
also, could this be a unique ptr rather than a shared ptr, or do we need to copy it somewhere?

Reply:
I adapted to the what was used for the similar interface (asyncForEachAttribute).
But looking at the usage now, I see no reason for it to be a shared_ptr.